packages/store/addon/-private/system/store.ts

```diff
@@ -2920,14 +2920,30 @@ const Store = Service.extend({
     return serializer;
   },
 
+  destroy() {
+    // enqueue destruction of any adapters/serializers we have created
+    for (let adapterName in this._adapterCache) {
+      let adapter = this._adapterCache[adapterName];
+      if (typeof adapter.destroy === 'function') {
+        adapter.destroy();
+      }
+    }
+
+    for (let serializerName in this._serializerCache) {
+      let serializer = this._serializerCache[serializerName];
+      if (typeof serializer.destroy === 'function') {
+        serializer.destroy();
+      }
+    }
+
+    return this._super();
+  },
+
   willDestroy() {
     this._super(...arguments);
     this._pushedInternalModels = null;
     this.recordArrayManager.destroy();
 
-    this._adapterCache = null;
-    this._serializerCache = null;
```

Comment:
@igorT these lines had already been removed by the time the original changes were made in https://github.com/emberjs/data/pull/7020 I had to also remove them to backport this change in order for the `destroy()` logic above to work. I'm not totally sure what, if any, implications this might have elsewhere. I found a comment related to this https://github.com/emberjs/data/pull/6207/files#r306956786 so I was thinking you might have more context

Reply:
👍on this PR, but I will double check with @runspired and @rwjblue  whether  this is still correct thinking